src/grasp_generator.cpp

```diff
@@ -1046,50 +1050,107 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
   }
-
+  grasp_poses.push_back(grasp_pose);
   addGrasp(grasp_pose, grasp_data, grasp_candidates, cuboid_top_pose, 0);
 
   // We define min, max and inc for each for loop here for readability
-  double x_min = -depth / 2.0 + grasp_data->active_suction_range_x_ / 2.0;
-  double x_max = depth / 2.0 - grasp_data->active_suction_range_x_ / 2.0;
-  double x_inc = grasp_data->grasp_resolution_;
 
-  double y_min = -width / 2.0 + grasp_data->active_suction_range_y_ / 2.0;
-  double y_max = width / 2.0 - grasp_data->active_suction_range_y_ / 2.0;
-  double y_inc = grasp_data->grasp_resolution_;
+  // if X range is less than y range then we use x range for the xy range
+  double xy_min, xy_max, xy_inc;
+  if (depth - grasp_data->active_suction_range_x_ < width - grasp_data->active_suction_range_y_)
+  {
+    xy_min = -depth / 2.0 + grasp_data->active_suction_range_x_ / 2.0;
+    xy_max = depth / 2.0 - grasp_data->active_suction_range_x_ / 2.0;
+    xy_inc = grasp_data->grasp_resolution_;
+  }
+  else
+  {
+    xy_min = -width / 2.0 + grasp_data->active_suction_range_y_ / 2.0;
+    xy_max = width / 2.0 - grasp_data->active_suction_range_y_ / 2.0;
+    xy_inc = grasp_data->grasp_resolution_;
+  }
 
-  double z_min = grasp_data->grasp_min_depth_;
-  double z_max = grasp_data->grasp_max_depth_;
+  double z_min = 0;
+  double z_max = grasp_data->grasp_max_depth_ - grasp_data->grasp_min_depth_;
   double z_inc = grasp_data->grasp_depth_resolution_;
 
   double yaw_min = 0.0;
   double yaw_max = 2.0 * M_PI;
   double yaw_inc = M_PI * (grasp_data->angle_resolution_ / 180.0);
 
-  for (double z = z_min; z <= z_max; z += z_inc)
+  // corner grasps at zero depth don't need variable angles
+  std::size_t num_grasps = grasp_poses.size();
+  for (std::size_t i = 0; i < num_grasps; ++i)
   {
-    for (double y = y_min; y <= y_max; y += y_inc)
+    for (double z = z_min; z <= z_max; z += z_inc)
     {
-      for (double x = x_min; x <= x_max; x += x_inc)
-      {
-        for (double yaw = yaw_min; yaw < yaw_max; yaw += yaw_inc)
-        {
-          Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
-          grasp_pose *= Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ());
-          grasp_pose.translation() = Eigen::Vector3d(x, y, z);
 
-          grasp_pose = cuboid_center_top_grasp.rotation() * grasp_pose;
-          grasp_pose.translation() += cuboid_center_top_grasp.translation();
+      Eigen::Affine3d grasp_pose = Eigen::Affine3d::Identity();
+      grasp_pose.translation() = Eigen::Vector3d(0, 0, z);
 
-          if (debug_top_grasps_)
-            visual_tools_->publishAxis(grasp_pose, rviz_visual_tools::MEDIUM, "after_transform");
+      grasp_pose = grasp_poses[i].rotation() * grasp_pose;
+      grasp_pose.translation() += grasp_poses[i].translation();
 
-          addGrasp(grasp_pose, grasp_data, grasp_candidates, cuboid_top_pose, 0);
-        }
-      }
+      grasp_poses.push_back(grasp_pose);
+    }
+  }
+
```

Comment:
add comments for these sections e.g. "Generate Y suction grasps" ?

Reply:
done